src/main/java/aqua/storage/Loadable.java

```diff
@@ -0,0 +1,14 @@
+package aqua.storage;
+
+import aqua.exception.LoadException;
+
+
+/** Represents an object whose state can be loaded. */
+public interface Loadable {
+    /**
+     * Loads the the state of this object from a data stored in hard disk.
+     *
+     * @throws LoadException if an error occured while loading.
+     */
+    public void load() throws LoadException;
```

Comment:
May I check the purpose of this class?

Reply:
Thank you @ChinJunAn for spotting this. I think this class will only be implemented by `AppManager` throughout this whole iP.

At first I thought there might be more things to load like maybe a Config class that stores the user preference (font size, path to save directory, etc) or a LogRecord class that stores the log records and so on. So then all of these classes will implement this Loadable interface and stored into a `List` where we use a for loop or Streams to laod all of them.

```java
for (Loadable loadable : loadables) {
    loadable.load();
}
```

```java
loadables.stream().forEach(loadable -> loadable.load());
```

The `try` and `catch` block has been ommitted and `loadables` is a `List<Loadable>` that has been initialised else where (maybe the constructor).

I think the benifits of this over line by line calling the `load()` method is that we can handle the LoadException easier. For example, if we want the program to continue loading even if one class fails to load then we can do something like this.

```java
List<LoadException> loadExs = new ArrayList<>();
loadables.stream()
        .forEach(loadable -> {
                try {
                    loadable.load();
                } catch (LoadException loadEx) {
                    // just add to list and continue with next
                    loadExs.add(loadEx);
                }
            });
if (!loadExs.isEmpty()) {
    for (LoadException loadEx : loadExs) {
        loadEx.printStackTrace();
    }
} else {
    System.out.println("Loaded everything successfully");
}
```

Again, for the scale of the project I do not think these will be used.
Thank you for your comment!!